addon/index.js

```diff
@@ -46,32 +45,32 @@ function tryScrollRecursively(fn, scrollHash) {
   })
 }
 
-let RouterScrollMixin = Mixin.create({
-  service: inject('router-scroll'),
+class EmberRouterScroll extends EmberRouter {
```

Comment:
Although this limits some flexibility, we will follow up with util functions that you can import and call in init to setup listeners as well if this pattern does not fix your app.

Reply:
This pattern - meaning extending from `EmberRouter`